sdk/core/core-tracing/lib/plugins/opencensus/openCensusTracePlugin.ts

```diff
@@ -0,0 +1,47 @@
+import { Tracer } from "../../interfaces/tracer";
+import { SpanOptions } from "../../interfaces/SpanOptions";
+import { Span } from "../../interfaces/span";
+import { OpenCensusSpanPlugin } from "../opencensus/openCensusSpanPlugin";
+
+export class OpenCensusTracePlugin implements Tracer {
+  private _tracer: any;
+
+  public constructor(tracer: any) {
+    this._tracer = tracer;
+  }
+
+  startSpan(name: string, options?: SpanOptions): Span {
+    const parent = options
+      ? options.parent
+        ? options.parent instanceof OpenCensusSpanPlugin
+          ? options.parent.getSpan()
+          : options.parent
+        : undefined
+      : undefined;
```

Comment:
What do we have to gain by doing a deep check like this vs 
`const parent = options && options.parent`

Reply:
This is because OpenCensus does not confirm to the OpenTelemetry Standards. So, you cannot pass options.parent (which is of type OpenCensusSpanPlugin which implements Span) It needs the specific span object created by OpenCensus Library

Reply:
If I understand this correctly, then you are saying that we need `options.parent.getSpan()` and not `options.parent`

The `parent` constant in the above code still points to only `options.parent` though...